templates/docker-container.yml

```diff
@@ -6,14 +6,35 @@
 trigger:
 - master
 
-pool:
-  vmImage: 'ubuntu-latest'
+resources:
+- repo: self
 
 variables:
-  imageName: 'your-container-image-name:$(build.buildId)'
-  dockerfile: '{{ dockerfilePath }}'
-  buildContext: '{{ buildContextPath }}'
+  # Container registry service connection established during pipeline creation - Docker Hub/ACR/...
+  dockerRegistryServiceConnection: '{{ containerRegistryConnection.Id }}'
+  imageRepository: '{{#toAlphaNumericString imageRepository 50}}{{/toAlphaNumericString}}'
+  containerRegistry: '{{ containerRegistryConnection.Authorization.Parameters.loginServer }}'
```

Comment:
What's the configuration panel going to look like for this template?

Are we asking server URL of the container registry as a text input for all container registries? Or for Azure do we show dropdowns for choosing the container registry based on the subscription chosen by the user (similar to AKS template)?